src/test/java/seedu/address/logic/parser/AddressBookParserTest.java

```diff
@@ -71,10 +78,33 @@ public void parseCommand_exit() throws Exception {
 
     @Test
     public void parseCommand_find() throws Exception {
-        List<String> keywords = Arrays.asList("foo", "bar", "baz");
-        FindCommand command = (FindCommand) parser.parseCommand(
-                FindCommand.COMMAND_WORD + " " + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FindCommand(new NameContainsKeywordsPredicate(keywords)), command);
+        List<String> nameKeywords = Arrays.asList("Assignment", "homework", "lab");
+
+        List<String> moduleCodeKeywords = Arrays.asList("CS2100", "CS2103T", "MA1101R");
+
+        List<String> deadlineKeywords = Arrays.asList("23-10-2020", "1200", "10-11-2021");
+
+        List<String> priorityKeywords = Arrays.asList("HIGH", "LOW");
+
+        FindCommand findByNames = (FindCommand) parser.parseCommand(
+                FindCommand.COMMAND_WORD + " " + PREFIX_NAME + nameKeywords.stream()
+                        .collect(Collectors.joining(" ")));
+
+        FindCommand findByModuleCodes = (FindCommand) parser.parseCommand(
+                FindCommand.COMMAND_WORD + " " + PREFIX_MODULE_CODE + moduleCodeKeywords.stream()
+                        .collect(Collectors.joining(" ")));
+
+        FindCommand findByDeadlines = (FindCommand) parser.parseCommand(
+                FindCommand.COMMAND_WORD + " " + PREFIX_DEADLINE + deadlineKeywords.stream()
+                        .collect(Collectors.joining(" ")));
+
+        FindCommand findByPriorities = (FindCommand) parser.parseCommand(
+                FindCommand.COMMAND_WORD + " " + PREFIX_PRIORITY + priorityKeywords.stream()
+                        .collect(Collectors.joining(" ")));
+        assertEquals(new FindCommand(new NameContainsKeywordsPredicate(nameKeywords)), findByNames);
+        assertEquals(new FindCommand(new ModuleCodeContainsKeywordsPredicate(moduleCodeKeywords)), findByModuleCodes);
+        assertEquals(new FindCommand(new DeadlineContainsKeywordsPredicate(deadlineKeywords)), findByDeadlines);
+        assertEquals(new FindCommand(new PriorityContainsKeywordsPredicate(priorityKeywords)), findByPriorities);
```

Comment:
Nice use of white spaces :) 

Though you may want to consider using the white spaces to not just improve readability but to group code with similar logic together (i.e grouping all the nameKeywords, deadlineKeywords... List<String> variables together and only inserting the white spaces in between priorityKeywords and findByNames. Likewise for the FindCommand variables and the assertEquals methods).

Reply:
Amended!